app/javascript/App/components/NewsfeedPage/NewsListItem.js

```diff
@@ -26,12 +27,53 @@ const NewsListItem = (props) => {
     .get('title')
     .replace(/<h1>/g, '')
     .replace(/<\/h1>/g, '')
+
   return (
     <div className={className} style={{ height: props.height || 'auto' }}>
       <div className="pa-default">
         <div
           className="pointer"
           onClick={() => {
+            const newsId = newsItem.get('id')
+            const readNewsData =
+              JSON.parse(localStorage.getItem('readNews')) || []
```

Comment:
Maybe leverage something like https://github.com/pamelafox/lscache instead of DIY?

Reply:
I like the recommendation! this lib looks pretty cool. i'm not sure how useful the mem-cache-like api will be for this particular use-case. the disadvantage is we can store less items because we store key, value, and time. if we omit storing time, then we lose the ability to expire the oldest items. my concern would be potentially degrading performance as we fill up the local storage faster than just storing the raw index.